contracts/colony/Colony.sol

```diff
@@ -271,6 +295,17 @@ contract Colony is ColonyStorage, PatriciaTreeProofs {
 
     // Add domain to local mapping
     initialiseDomain(newLocalSkill);
+
+    if (keccak256(abi.encodePacked(_metadata)) != keccak256(abi.encodePacked(""))) {
+      emit DomainMetadata(domainCount, _metadata);
+    }
+  }
+
+  function editDomain(uint256 _permissionDomainId, uint256 _childSkillIndex, uint256 _domainId, string memory _metadata) public
+  stoppable
+  authDomain(_permissionDomainId, _childSkillIndex, _domainId)
+  {
+    emit DomainMetadata(_domainId, _metadata);
```

Comment:
For instance, what if they pass `""` here?

Reply:
Fair, I'll put the check here too

Reply:
But, what if they want to delete the metadata?

Reply:
Then they should be passing the IPFS hash of the empty JSON object.

Reply:
Actually, I'm not sure I should put the check here. There's no reason to call this function if you're not emitting metadata, which is the case for the functions where I have the check.

Reply:
I don't feel so strongly I suppose, I'll just say that if they can pass the empty string, then the app must be able to handle the empty string, which makes me less worried about emitting it in the other cases. I see the argument for gas saving, but it makes the code more complex... it's whatever you like I guess.

Reply:
I came down on the side of adding the guard here, for consistency more than anything else. You're absolutely right that the app has to sanitize the inputs here. I certainly expect a `validIPFSHash` or similar to appear in the app, alongside other validation functions though, which would be able to easily handle the special case of the empty string being passed.